addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -100,11 +100,15 @@ export class PaymentScreen extends Component {
     get selectedPaymentLine() {
         return this.currentOrder.selected_paymentline;
     }
-    addNewPaymentLine(paymentMethod) {
+    addNewPaymentLine(paymentMethod, automatic_payment) {
         // original function: click_paymentmethods
-        const result = this.currentOrder.add_paymentline(paymentMethod);
+        const result = this.currentOrder.add_paymentline(paymentMethod, automatic_payment);
         if (result) {
             this.numberBuffer.reset();
+            if (automatic_payment) {
+                const newPaymentLine = this.paymentLines[this.paymentLines.length - 1];
```

Comment:
```suggestion
                const newPaymentLine = this.paymentLines.at(-1);
```
No need to change this now; just know that this is an option